servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/ContentCodec.java

```diff
@@ -41,19 +41,18 @@
      * @return {@link Buffer} the result buffer with the content encoded
      */
     default Buffer encode(Buffer src, BufferAllocator allocator) {
-        return encode(src, src.readerIndex(), src.readableBytes(), allocator);
+        return encode(src, src.readableBytes(), allocator);
     }
 
     /**
      * Take a {@link Buffer} and encode its contents resulting in a {@link Buffer} with the encoded contents.
      *
      * @param src the {@link Buffer} to encode
-     * @param offset the offset of the source to start reading from
      * @param length the total length available for reading
      * @param allocator the {@link BufferAllocator} to use for allocating auxiliary buffers or the returned buffer
      * @return {@link Buffer} the result buffer with the content encoded
      */
-    Buffer encode(Buffer src, int offset, int length, BufferAllocator allocator);
+    Buffer encode(Buffer src, int length, BufferAllocator allocator);
```

Comment:
It is more standard API, but in our case it seems (to me) more confusing.
Buffer has the reader/writer indexes, which these method mutate. So we typically mutate the readerIndex after consuming the buffer, to `current_readerIndex + limit`. If we support `offset`, that can be different than the `readerIndex`, (before or after), what happens then to the readerIndex if the `offset + limit` < `current_readerIndex`? Do we move it back to adhere to the contract or keep it unchanged?
I think since the `Buffer` API allows easy positioning of the readerIndex, we should allow the caller to rely on that for offsetting.

WDYT?

Reply:
Good point. Let's discuss offline what should be the default behavior of all codecs and deserializers. We are currently inconsistent. If we will decide to move indexes everywhere, then having only `length` here is reasonable. If we decide not to move indexes, then `offset` may be important.